### PR TITLE
File import via Param YML folder using http request

### PR DESCRIPTION
**Pull request by jacobchristensen79**

### CODE REVIEW

File: src/Command/ImportCommand.php [hunk 1]
```diff
@@ -321,4 +324,29 @@ protected function parseContent($content, $file)
...
+    private function validateUploadFile(Document $doc, $originFile)
+    {
+        $documentData = $doc->getData();
+
+        if (!array_key_exists('file', $documentData)) {
+            return false;
+        }
+
+        // Find file
+        $fileName = preg_replace('/([^\/]+$)/', $documentData['file'], $originFile);
```
Comment (hairmare) [inline src/Command/ImportCommand.php]: What do this and the next line do that [`basename()`](http://php.net/manual/en/function.basename.php) can't solve?

The longer I look at these lines the less I get what they're trying to achieve...

File: src/Command/ImportCommandAbstract.php [hunk 2]
```diff
@@ -47,14 +48,27 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $files = $input->getArgument('file');
         $finder = $this->finder->files();
 
+        /**
+         * @param SplFileInfo $file
+         * @return bool
+         */
+        $filter = function (SplFileInfo $file) {
```
Comment (hairmare) [inline src/Command/ImportCommandAbstract.php]: This feature (as well as the ignoring of dotfiles) needs some tests to ensure that it does what's intended.

My comment on the previous PR was not about the feature but about the lack of tests.

File: src/Service/HttpClient.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        $options = $this->checkFileUploadRequest($options);
+
+        return parent::requestAsync($method, $this->url, $options);
+    }
+
+    /**
+     * @param array $options Curl data options
+     * @return array options
+     */
+    private function checkFileUploadRequest($options)
```
Comment (hairmare) [inline src/Service/HttpClient.php]: This **method** needs to be tested. Actually, the whole class deserves tests and might need a slight refactor to make it testable in the first place.

File: src/Command/ImportCommand.php (hunk 1 above)
Comment (jacobchristensen79) [inline src/Command/ImportCommand.php]: if between --- and --- there is NO line file then do not try to check if there is a file....means, no file to be uploaded. Make this back compatible with current yml files.

File: src/Service/HttpClient.php (hunk 3 above)
Comment (jacobchristensen79) [inline src/Service/HttpClient.php]: This is tested, please view the tests are updated to use this method.

File: src/Command/ImportCommandAbstract.php (hunk 2 above)
Comment (jacobchristensen79) [inline src/Command/ImportCommandAbstract.php]: Again, tests have been updated and also testing sending a file. please review with knowledge or just decline it all

File: src/Service/HttpClient.php (hunk 3 above)
Comment (hairmare) [inline src/Service/HttpClient.php]: Please point me to the tests then. Scrutinizer [does not see any tests](https://scrutinizer-ci.com/g/libgraviton/import-export/inspections/7e988a71-628d-4956-9425-d4373aa8e309/code-structure/class/Graviton%5CImportExport%5CService%5CHttpClient?expandCoverage=1).

File: src/Command/ImportCommand.php (hunk 1 above)
Comment (hairmare) [inline src/Command/ImportCommand.php]: But that already gets caught by the `return false` above. What is wrong with `basename()`?

File: src/Command/ImportCommandAbstract.php (hunk 2 above)
Comment (hairmare) [inline src/Command/ImportCommandAbstract.php]: They don't test skipping a non `yml` or `json` file, we don't even have fixtures to test that.

File: src/Service/HttpClient.php (hunk 3 above)
Comment (jacobchristensen79) [inline src/Service/HttpClient.php]: https://github.com/libgraviton/import-export/blob/feature/EVO-3237-send-file-parser/test/Command/ImportCommandTest.php#L28
and all tests here are using this Mock.

File: src/Command/ImportCommandAbstract.php (hunk 2 above)
Comment (jacobchristensen79) [inline src/Command/ImportCommandAbstract.php]: If who created in first instance this importer did not tests what he did:
https://github.com/libgraviton/import-export/blob/feature/EVO-3237-send-file-parser/src/Command/ImportCommand.php#L306
so, yes, we escape yml vs json the "hard" way. cheers

Comment (jacobchristensen79) [inline src/Command/ImportCommandAbstract.php]: If I do not filter in "finder" then I have to hardcode in the mentioned function that we "allow" all file types but do not try to import them. This work around is a lot more complicated, for not mentioned pointless.

File: src/Command/ImportCommand.php [hunk 4]
```diff
@@ -341,7 +341,7 @@ private function validateUploadFile(Document $doc, $originFile)
         }
 
         // Find file
-        $fileName = preg_replace('/([^\/]+$)/', $documentData['file'], $originFile);
+        $fileName = dirname($originFile) . DIRECTORY_SEPARATOR . $documentData['file'];
         $fileName = str_replace('//', '/', $fileName);
```
Comment (jacobchristensen79) [inline src/Command/ImportCommand.php]: Now, I hope you like this. But this also means that the file MUST be placed in exact same folder.
before you could have a extern folder and refer to the file with ../../../myImageFolder/img.png
Now you need files to be in same folder.

**Comment by jacobchristensen79** [comment]
I close this so someone can do it so we are all happy

**Pull request closed without merging**